server/server.js

```diff
@@ -142,24 +146,49 @@ Server.prototype._handleClientMessage = function(client, data) {
     return;
   }
 
-  logging.info('#client.message - received', client.id, message,
-     (this.resources[message.to] ? 'exists' : 'not instantiated'),
-     (this.subs[message.to] ? 'is subscribed' : 'not subscribed')
-    );
+  if (!this._clientDataPersist(client.id, message)) {
+    return;
+  }
+
+  this._resourceMessageHandle(client, message);
+};
+
+// Initialize a client, and persist messages where required
+Server.prototype._clientDataPersist = function (id, message) {
+  // Sync the client name to the current client id
+  if (message.op == 'name_id_sync') {
+    this._clientInit(message);
+    this.clientVersion = message.client_version;
```

Comment:
we probably need not store clientVersion on the server? (It will keep changing based on the client and the message). Here, can't we just use a local (stack) var instead?